src/wp-includes/class-wp-image-editor-gd.php

```diff
@@ -323,7 +323,13 @@ public function crop( $src_x, $src_y, $src_w, $src_h, $dst_w = null, $dst_h = nu
 			$dst_h = $src_h;
 		}
 
-		$dst = wp_imagecreatetruecolor( $dst_w, $dst_h );
+		foreach( array( $src_x, $src_y, $src_w, $src_h, $dst_w, $dst_h ) as $value ) {
+			if ( ! is_numeric( $value ) || (int) $value <= 0 ) {
+				return new WP_Error( 'image_crop_error', __( 'Image crop failed.' ), $this->file );
```

Comment:
It'd be good have add some unit tests for this.

Reply:
@iandunn Tests added in commit https://github.com/WordPress/wordpress-develop/pull/791/commits/c7cf551b35e0cb4329cace5e133c10d476eee0b8